config/config/sc-config.yaml

```diff
@@ -328,6 +328,15 @@ thanos:
     # They are infinitly scalable
     replicaCount: 1
 
+    topologySpreadConstraints:
+    - labelSelector:
+        matchLabels:
+          app.kubernetes.io/name: thanos
+          app.kubernetes.io/component: query
+      maxSkew: 1
+      topologyKey: kubernetes.io/hostname
+      whenUnsatisfiable: DoNotSchedule
```

Comment:
I feel that it is a bit harsh to have a hard requirement to not run multiple queries on the same node since it's a stateless component.

Reply:
@Xartos @crssnd @aarnq What do you think about this?

Reply:
I'd prefer to force them to be scheduled on separate nodes, as they are used by the rulers, though via the frontend which is a single point of failure...
And with a heavy query two replicas can by default eat ~2Gi of pure overprovisioned memory, which I think we should distribute to limit the potential impact.

Reply:
Yea, I kind of agree that having them spread apart is quite important.
If we had some smarter scheduler that would put it there and eventually would schedule it on a separate node I would be fine with it. But since we want the replicas for redundancy it feel like this is the way to go